app/models/pipeline_run.rb

```diff
@@ -1570,16 +1650,27 @@ def outputs_by_step(can_see_stage1_results = false)
       num_steps = targets.length
       targets.each_with_index do |(target_name, output_list), step_idx|
         next if given_targets.keys.include?(target_name)
-        file_info = []
+
+        file_paths = []
         output_list.each do |output|
-          file_info_for_output = filename_to_info["#{output_dir_s3_key}/#{pipeline_version}/#{output}"]
+          file_paths << "#{output_dir_s3_key}/#{pipeline_version}/#{output}"
+        end
+        if prs.name == PipelineRunStage::HOST_FILTERING_STAGE_NAME && target_name == 'star_out'
```

Comment:
This is the wrong place for this logic. The whole point of `output_states` is that they encode the outputs that are supposed to get loaded. Please leave `outputs_by_step` untouched and find a way to work within the boundaries of the design philosophy?

Reply:
Is there a way to extend the framework with a way of capturing these "optional" files rather than hardcoding this stuff ad-hoc based on name of the step/stage?

Reply:
There are two possibilities: 
1) configuration: have an extra field name output_fields that tells the client what they should look for.
2) output: each stage outputs a status file (e.g. `samples/<project_id>/<sample_id>/results/3.17/host_filter_status.json`) that contains information about the execution. You can include there the list of `additional_files_to_uload` and the `optional_files_to_upload` that the pipeline step generates.

I believe the second option is the best...
In the end you can include those files here. @cdebourcy do you know of any privacy limitations on files that pipeline steps add to these two arrays?

Reply:
Thanks @tfrcarvalho . This is also relevant to my PR https://github.com/chanzuckerberg/idseq-dag/pull/251 .